draft-ietf-ppm-dap.md

```diff
@@ -789,6 +795,11 @@ provide for some small leeway, usually no more than a few minutes, to account
 for clock skew. If the leader rejects a report for this reason, it SHOULD abort
 the upload protocol and alert the client with error "reportTooEarly".
 
+The Leader SHOULD ignore any reports pertaining to a task that have timestamps pass
+`max_task_lifetime`. Leader SHOULD abort the upload protocol and alert the client
+with error "reportTooLate". Client MAY choose to opt out of the task if its own clock
+has passed `max_task_lifetime`.
```

Comment:
Should the leader make this determination based on the timestamp in the report, or based on the leader's perception of the current time when it receives the report?

Put another way: suppose a task has `max_batch_lifetime` = `t_1`. The leader receives a report with timestamp `t_0 < t_1`. The leader checks its own clock and determines that the current time is `t_2 > t_1`. Should the report be rejected because it was too late by the time the leader saw it, or accepted because its timestamp indicates the measurement was taken before the end of the task?

I think it's got to be the former (because otherwise clients could send back-dated reports long after a task has expired), in which case this should be rewritten to discuss time of report reception rather than the timestamp.

Reply:
@tgeoghegan I think it should be based on a timestamp in the report. Assuming we want to give client a chance to check this, it can't be a duration since client may "wake up" at very different times.
Even for aggregators, if we determine by arrival time, then we could lose reports due to any latency caused on route to leader. The threat you mentioned is real, but attacker would have to hijack a large number of clients to make this attack effective.

Reply:
having said that, I realised that given the `time_precision` is used to blur client timestamp, if we really want client to check this expiry time, then we should round (up?) `task_expiry` with `time_precision` too, otherwise if time_expiry is close to the rounded timestamp, people could figure out a more accurate timestamp of client.

Reply:
I think you're right that it makes sense to trust the timestamp in the report.

Can you expand on why evaluating `task_expiration` could leak information about the report? Which actor (the client, the aggregator, a network observer) could learn something about the true time a measurement was taken based on what observable behavior of which actor?

Reply:
@tgeoghegan for e.g. if the precision is in 1hr, and 
1. one client has rounded it's true timestamp to 14:00:00
2. the aggregators know the `task_expiration` is effectively 14:00:01 UTC
3. the aggregators know clients will check their timestamp < `task_expiration` before sending.

Then the aggregator could know that the client has a true timestamp between 14:00:00 and 14:00:01.

Reply:
Can we resolve this convo?

Reply:
I see what you mean @wangshan. I think requiring that `task_expiration` be aligned with `time_precision` would help. Would you please file an issue about that so we can deal with it in another PR?

Reply:
will do @tgeoghegan, let me first resolve this conversation and rebase this PR for merging